sentry/src/db/campaign.rs

```diff
@@ -49,6 +49,51 @@ pub async fn fetch_campaign(pool: DbPool, campaign: &Campaign) -> Result<Campaig
     Ok(Campaign::from(&row))
 }
 
+pub async fn get_campaigns_for_channel(pool: &DbPool, campaign: &Campaign) -> Result<Vec<Campaign>, PoolError> {
+    let client = pool.get().await?;
+    let statement = client.prepare("SELECT id, channel, creator, budget, validators, title, pricing_bounds, event_submission, ad_units, targeting_rules, created, active_from, active_to FROM campaigns WHERE channel_id = $1").await?;
+
+    let row = client.query(&statement, &[&campaign.channel.id()]).await?;
```

Comment:
you also have `query_one` which will return an error if the query returns 0 or >1 results.

Reply:
In this case, I'm trying to get all campaigns for the same channel with one query.

Reply:
can you use a plural word - `rows`.